actors/miner/src/lib.rs

```diff
@@ -1950,6 +1951,116 @@ impl Actor {
+                sealed_cid: sector.sealed_cid,
+                unsealed_cid: CompactCommD::empty().get_cid(params.seal_proof_type).unwrap(),
+            };
+            rt.batch_verify_ni_seals(&vec![seal_verify_info])
+                .context_code(ExitCode::USR_ILLEGAL_ARGUMENT, "failed to batch verify")?;
+
+            let duration = sector.expiration - activation_epoch;
+            let power =
+                qa_power_for_weight(info.sector_size, duration, &BigInt::zero(), &BigInt::zero());
+
+            let rew = request_current_epoch_block_reward(rt)?;
+            let pwr = request_current_total_power(rt)?;
+            let circulating_supply = rt.total_fil_circ_supply();
+            let pledge_inputs = NetworkPledgeInputs {
+                network_qap: pwr.quality_adj_power_smoothed,
+                network_baseline: rew.this_epoch_baseline_power,
+                circulating_supply,
+                epoch_reward: rew.this_epoch_reward_smoothed,
+            };
+
+            let storage_pledge = expected_reward_for_power(
+                &pledge_inputs.epoch_reward,
+                &pledge_inputs.network_qap,
+                &power,
+                INITIAL_PLEDGE_PROJECTION_PERIOD,
+            );
+
+            let initial_pledge = initial_pledge_for_power(
+                &power,
+                &pledge_inputs.network_baseline,
+                &pledge_inputs.epoch_reward,
+                &pledge_inputs.network_qap,
+                &pledge_inputs.circulating_supply,
+            );
+
+            let day_reward = expected_reward_for_power(
+                &pledge_inputs.epoch_reward,
+                &pledge_inputs.network_qap,
+                &power,
+                fil_actors_runtime::EPOCHS_IN_DAY,
+            );
+            total_pledge += &initial_pledge;
+
+            let new_sector_info = SectorOnChainInfo {
+                sector_number: sector.sector_number,
+                seal_proof: params.seal_proof_type,
+                sealed_cid: sector.sealed_cid,
+                deprecated_deal_ids: vec![],
+                expiration: sector.expiration,
+                activation: activation_epoch,
+                deal_weight: DealWeight::zero(),
+                verified_deal_weight: DealWeight::zero(),
+                initial_pledge: initial_pledge,
+                expected_day_reward: day_reward,
+                expected_storage_pledge: storage_pledge,
+                power_base_epoch: activation_epoch,
+                replaced_day_reward: TokenAmount::zero(),
+                sector_key_cid: None,
+                flags: SectorOnChainInfoFlags::SIMPLE_QA_POWER,
+            };
+
+            sectors_to_add.push(new_sector_info);
+        }
+
+        rt.transaction(|state: &mut State, rt| {
+            state.put_sectors(store, sectors_to_add).map_err(|e| {
+                e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to put new sectors")
+            })?;
```

Comment:
Avoid `map_err` and `downcast_default`. Use `with_context_code` and similar instead for new code.